### PR TITLE
Add details about x509 configuration to javadoc.

### PR DESCRIPTION
**Pull request by boaks**
Signed-off-by: Achim Kraus <achim.kraus@bosch-si.com>

### CODE REVIEW

File: scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java [hunk 1]
```diff
@@ -1436,12 +1437,19 @@ public Builder setIdentity(PrivateKey privateKey, PublicKey publicKey) {
 		 * 
 		 * @param privateKey the private key used for creating signatures
 		 * @param certificateChain the chain of X.509 certificates asserting the
-		 *            private key subject's identity
+		 *            private key subject's identity. The endpoints certificate
+		 *            must be at position 0, the certificate signed by a trusted
+		 *            CA must be a t the highest position. A self-signed
```
Comment (sbernard31) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java]: typo : "a t"

File: scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java [hunk 2]
```diff
@@ -1436,12 +1437,19 @@ public Builder setIdentity(PrivateKey privateKey, PublicKey publicKey) {
 		 * 
 		 * @param privateKey the private key used for creating signatures
 		 * @param certificateChain the chain of X.509 certificates asserting the
-		 *            private key subject's identity
+		 *            private key subject's identity. The endpoints certificate
```
Comment (sbernard31) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java]: endpoint~s~ ?

File: scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java [hunk 3]
```diff
@@ -1482,12 +1490,19 @@ public Builder setIdentity(PrivateKey privateKey, Certificate[] certificateChain
...
+		 *            must be at position 0, the certificate signed by a trusted
+		 *            CA must be a t the highest position. A self-signed
+		 *            top-level certificate will be removed for outgoing
+		 *            {@link CertificateMessage}. If used for a client side
+		 *            {@link CertificateMessage}, the chain will be truncated to
+		 *            the first certificate of one of the received certificate
+		 *            authorities.
```
Comment (sbernard31) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java]: same typo here

File: scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java [hunk 4]
```diff
@@ -1543,11 +1558,17 @@ public Builder setIdentity(PrivateKey privateKey, Certificate[] certificateChain
 		 * requesting a client certificate during the DTLS handshake.</li>
 		 * </ul>
 		 * 
+		 * When a RFC5250 compliant verification is required, use use only
```
Comment (sbernard31) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java]: use ~use~

File: scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java (hunk 1 above)
Comment (boaks) [reply scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java]: OK. Fixed.

File: scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java (hunk 2 above)
Comment (boaks) [reply scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java]: OK. changed to "endpoint's".

File: scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java (hunk 4 above)
Comment (boaks) [reply scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java]: OK. fixed.

**Pull request merged**